### PR TITLE
Angular 7/8 upgrade, audit fixes

### PR DESCRIPTION
**Pull request by dhlevi**
Upgrade to from Angular 6 to 8

asdf config .tools is moving from node 8 to 10.19, which should proof us for an eventual move to node 9. If you use nvm instead of asdf just manually toggle before you build!

Be sure to do a clean install/rebuild, and due to the node change you may have to run "npm rebuild node-sass"

Because of the number of changes, I'm including everyone. Note that once this is in place, any outstanding PR's will need some careful merging!

See tickets [EE-894](https://bcmines.atlassian.net/browse/EE-894) and [EE-930](https://bcmines.atlassian.net/browse/EE-930)

### CODE REVIEW

File: src/app/projects/add-edit-project/form-tab-2018/form-tab-2018-component.spec.ts [hunk 1]
```diff
@@ -1,6 +1,6 @@
-import { async, ComponentFixture, TestBed } from '@angular/core/testing';
-import { NO_ERRORS_SCHEMA } from '@angular/core';
-import { FormTab2018Component } from './form-tab-2018.component';
+// import { async, ComponentFixture, TestBed } from '@angular/core/testing';
```
Comment (KitArmstrong) [inline src/app/projects/add-edit-project/form-tab-2018/form-tab-2018-component.spec.ts]: Can these be dropped?

Comment (dhlevi) [reply src/app/projects/add-edit-project/form-tab-2018/form-tab-2018-component.spec.ts]: Yes. I would like to replace the dud test with a real one at some point, but that's probably not the point of the upgrade exercise.

File: src/app/services/auth-guard.service.ts [hunk 2]
```diff
@@ -11,7 +11,8 @@ export class AuthGuard implements CanActivate {
     private keycloakService: KeycloakService, ) {
   }
 
-  canActivate(next: ActivatedRouteSnapshot, state: RouterStateSnapshot): Observable<boolean> | Promise<boolean> | boolean {
+  // canActivate(next: ActivatedRouteSnapshot, state: RouterStateSnapshot): Observable<boolean> | Promise<boolean> | boolean {
```
Comment (mtCarto) [inline src/app/services/auth-guard.service.ts]: Can we remove this?

File: src/app/services/recent-activity.ts [hunk 3]
```diff
@@ -11,41 +10,6 @@ export class RecentActivityService {
...
-  //       return [];
-  //     })
-  //     .catch(error => this.api.handleError(error));
-  // }
-
-  // deleteByApplicationId(applicationId: string): Observable<Object> {
-  //   return this.api.deleteRecentActivitysByApplicationId(applicationId)
-  //     .catch(error => this.api.handleError(error));
-  // }
-
```
Comment (mtCarto) [inline src/app/services/recent-activity.ts]: Cruft cleanup 💯

**Pull request merged**